File: content/snippets/cpp/language_basics_maps_and_sets.cpp
#include <iostream>
#include <map>
#include <set>
#include <string>

int main() {
    std::map<std::string, int> counts;
    counts["apple"] = 2;
    counts["banana"] = 1;
    counts["apple"] += 3;

    std::set<std::string> tags{"fruit", "food"};
    tags.insert("fresh");

    std::cout << "apple count: " << counts["apple"] << "\n";

    std::cout << "all map entries: ";
    for (const auto& p : counts) {
        std::cout << "(" << p.first << "," << p.second << ") ";
    }
    std::cout << "\n";

    std::cout << "all set entries: ";
    for (const auto& t : tags) {
        std::cout << t << ' ';
    }
    std::cout << "\n";
    return 0;
}

File: content/snippets/go/language_basics_maps_and_sets.go
package main

import "fmt"

func main() {
	counts := map[string]int{
		"apple":  2,
		"banana": 1,
	}
	counts["apple"] += 3

	tags := map[string]struct{}{
		"fruit": {},
		"food":  {},
	}
	tags["fresh"] = struct{}{}

	fmt.Println("apple count:", counts["apple"])
	fmt.Println("all map entries:")
	for k, v := range counts {
		fmt.Printf("(%s,%d) ", k, v)
	}
	fmt.Println()

	fmt.Println("all set entries:")
	for t := range tags {
		fmt.Printf("%s ", t)
	}
	fmt.Println()
}
